### PR TITLE
Add a new type of task, JiraUpdateTask.

### LINKED ISSUE #18: Addition of a new type of task that updates JIRA issues
We already have a task that transitions JIRA issues from one state to another, possibly with field updates, but we don't have one that will _only_ update fields. We should add this so that we can build more complex JIRA add-ons.

### PR DESCRIPTION
**Pull request by jwir3**
Fixes #18.

### CODE REVIEW

File: gruel/src/main/groovy/com/glasstowerstudios/gruel/tasks/jira/JiraUpdateTask.groovy [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    this.jql = jql
+  }
+
+  void setFieldUpdates(LinkedHashMap<String, Object> fieldUpdates) {
+    this.fieldUpdates = fieldUpdates
+  }
+
+  @TaskAction
+  def doTask() {
+    def credentials = new BasicCredentials(userName, password)
```
Comment (abdyer) [inline gruel/src/main/groovy/com/glasstowerstudios/gruel/tasks/jira/JiraUpdateTask.groovy]: Would it be useful to create a base `JiraTask` to centralize some of this common logic duplicated in `JiraTransitionTask` and `JiraUpdateTask`?

**Review by abdyer — COMMENTED** [review]
LGTM other than possibly centralizing some of the common Jira client stuff.

File: gruel/src/main/groovy/com/glasstowerstudios/gruel/tasks/jira/JiraUpdateTask.groovy (hunk 1 above)
Comment (jwir3) [reply gruel/src/main/groovy/com/glasstowerstudios/gruel/tasks/jira/JiraUpdateTask.groovy]: Yep, let me do some refactoring. I'm working with something similar for Github issue(s), and I'll use that model.

**Comment by jwir3** [comment]
@abdyer Ready for review of the new stuff. Once this and the github integration are complete, I'll push a 0.0.7-SNAPSHOT build to maven.

File: gruel/src/main/groovy/com/glasstowerstudios/gruel/GruelPlugin.groovy [hunk 2]
```diff
@@ -16,6 +17,7 @@ public class GruelPlugin implements Plugin<Project> {
     gruelExtension.setProject(aProject);
 
     aProject.extensions.create("hipchat", HipChatExtension);
+    def jiraExtension = aProject.extensions.create("jira", JiraExtension);
```
Comment (abdyer) [inline gruel/src/main/groovy/com/glasstowerstudios/gruel/GruelPlugin.groovy:20]: Nice. Didn't know about this.

**Review by abdyer — APPROVED** [review]
So fresh and so clean. :shipit:

File: gruel/src/main/groovy/com/glasstowerstudios/gruel/GruelPlugin.groovy (hunk 2 above)
Comment (jwir3) [reply gruel/src/main/groovy/com/glasstowerstudios/gruel/GruelPlugin.groovy:20]: Yeah, I'm actually going to make it part of the Gruel extension in a future modification.

**Pull request merged**